views/course_form/views.py

```diff
@@ -38,6 +42,63 @@ def post(self, request, code: str | None = None, semester: str | None = None):
         if not self.__can_use_form(request.user, code, semester):
             return redirect(reverse("home"))
 
+            # Handle delete action
+        if request.POST.get('action') == 'delete':
+            # Check if the user is an admin
+            if request.user.role != "Admin":
+                return redirect(reverse("home"))  # Redirect non-admins to home
+            CourseController.delete_course(code, semester)
+            return redirect(reverse("home"))
+
+        #validation checks
+        #gather input data
+        course_code = request.POST.get('course_code', '').strip()
+        course_name = request.POST.get('course_name', '').strip()
+        selected_semester = request.POST.get('semester', '').strip()
+
+        # Validation errors dictionary
+        errors = {
+            "course_code": "",
+            "course_name": "",
```

Comment:
why don't you just make this a list of values that you push values onto. That would make it more extensible to other errors in the future.